level-1/문자열-다루기-기본&12918&.js

```diff
@@ -43,4 +43,19 @@ function solution(s) {
   // 위의 모든 조건에 포함되지 않으면
   // (길이가 4 혹은 6이고, 숫자로만 구성되어 있으면) true 반환
   return true;
-}
\ No newline at end of file
+}
+
+//정답 5 - soulhn
+function solution(s) {
+  // 길이가 4이거나 6인지 확인
+  if (s.length === 4 || s.length === 6) {
+    //배열로 변환, Int로 변환, 배열 내부에 NaN이 있는지 확인 
+    //NaN이 존재하면 true 반환이기에 ! 이용하여 false 반환
+    return !s
+      .split("")
+      .map((i) => parseInt(i, 10))
+      .includes(NaN);
```

Comment:
NaN 이 있느냐를 확인하시는 것이라면, `some` 메서드를 사용해보셔도 좋을 것 같아요!
 
```suggestion
      .some(i => isNaN(parseInt(i, 10)));
```

[MDN some](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Array/some)